rtos/TARGET_CORTEX/mbed_rtx_idle.cpp

```diff
@@ -47,7 +48,11 @@ extern "C" {
     {
         // Do not use SingletonPtr since this relies on the RTOS
         if (NULL == os_timer) {
-            os_timer = new (os_timer_data) rtos::internal::SysTimer();
+#if MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER
```

Comment:
Should this also be automatically inferred if `DEVICE_LPTICKER` is unset, rather than needing to be set manually?

    #if !DEVICE_LPTICKER
    #  define USE_US_TICKER true
    #else 
    #  define USE_US_TICKER MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER
    #endif

maybe.

Reply:
If it is automatically inferred then `MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER` may be false (default) yet the microsecond ticker is used. This could lead to problems if `MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER` is used elsewhere. I added compiler asserts so you get a clear error message if the required DEVICE_*TICKER is missing.